src/main/java/com/labsynch/labseer/chemclasses/rdkit/BBChemStructureService.java

```diff
@@ -0,0 +1,338 @@
+import org.RDKit.RWMol;
+import org.RDKit.SanitizeFlags;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ArrayNode;
+import org.codehaus.jackson.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+@Component
+public class BBChemStructureService  implements ExternalStructureService {
+    
+    Logger logger = LoggerFactory.getLogger(BBChemStructureService.class);
+
+	static {
+		System.loadLibrary("GraphMolWrap");
+	}
+
+	@Autowired
+	private PropertiesUtilService propertiesUtilService;
+
+	
+	@Override
+    public void populateDescriptors(RDKitStructure rdKitStructure) {
+
+		RWMol mol = getPartialiallySanizedRWMol(rdKitStructure.getMol());
+
+		// https://www.rdkit.org/docs/cppapi/classRDKit_1_1ROMol.html#a48ab4e1692b503eda6d31c5bb10fe0bb
+		// calculates any of our lazy properties
+		// 	bool 	strict = true	
+		mol.updatePropertyCache(false);
+		
+		// https://www.rdkit.org/docs/cppapi/namespaceRDKit_1_1Descriptors.html#a4a0f6886e3798fcb195b6c270a3645c6
+		// const ROMol & 	mol,
+		// bool 	onlyHeavy = false
+		rdKitStructure.setAverageMolWeight(RDKFuncs.calcAMW(mol, false));
+
+		// https://www.rdkit.org/docs/cppapi/namespaceRDKit_1_1Descriptors.html#a468168875d98b415e14c6eac9d7c36ae
+		// const ROMol & 	mol,
+		// bool  onlyHeavy = false
+		// https://www.mail-archive.com/rdkit-discuss@lists.sourceforge.net/msg10222.html
+		// MolWt uses naturally occurring average atomic weights, the ones you find in
+		// a typical periodic table. For example, Cl = 35.453.
+		// ExactMolWt uses the weight of a specific isotope (the most naturally
+		// abundant isotope unless the structure specifies a different one for an
+		// atom). These are the atomic weights you find in an isotope chart, not a
+		// regular periodic table. For example, 35Cl = 34.96885268.
+		// https://www.nist.gov/pml/atomic-weights-and-isotopic-compositions-relative-atomic-masses
+		rdKitStructure.setExactMolWeight(RDKFuncs.calcExactMW(mol, false));
+
+		// https://www.rdkit.org/docs/cppapi/namespaceRDKit.html#a75e1240acc451a7a73280220f79b6c05
+		// mol	: the molecule in question.
+		// doIsomericSmiles	: include stereochemistry and isotope information in the SMILES
+		// doKekule	: do Kekule smiles (i.e. don't use aromatic bonds) NOTE that this will throw an exception if the molecule cannot be kekulized.
+		// rootedAtAtom	: make sure the SMILES starts at the specified atom. The resulting SMILES is not, of course, canonical.
+		// canonical	: if false, no attempt will be made to canonicalize the SMILES
+		// allBondsExplicit	: if true, symbols will be included for all bonds.
+		// allHsExplicit	: if true, hydrogen counts will be provided for every atom.
```

Comment:
This is very helpful. Seems to be missing the "doRandom" flag whose behavior is undocumented on that RDKit doc link. But I think it is better to have every argument accounted for even if we're not sure what this last one does.